src/virtual_secondary/managedsecondary.cc

```diff
@@ -139,44 +142,13 @@ bool ManagedSecondary::putRoot(const std::string &root, const bool director) {
   return true;
 }
 
-bool ManagedSecondary::sendFirmware(const std::string &data) {
-  std::lock_guard<std::mutex> l(install_mutex);
-
-  if (expected_target_name.empty()) {
-    return false;
-  }
-  if (!detected_attack.empty()) {
-    return false;
-  }
-
-  if (data.size() > static_cast<size_t>(expected_target_length)) {
-    detected_attack = "overflow";
-    return false;
+data::ResultCode::Numeric ManagedSecondary::install(const Uptane::Target &target_name) {
+  if (fiu_fail((std::string("secondary_install_") + getSerial().ToString()).c_str()) != 0) {
+    return data::ResultCode::Numeric::kInstallFailed;
   }
 
-  std::vector<Uptane::Hash>::const_iterator it;
-  for (it = expected_target_hashes.begin(); it != expected_target_hashes.end(); it++) {
-    if (it->TypeString() == "sha256") {
-      if (boost::algorithm::to_lower_copy(boost::algorithm::hex(Crypto::sha256digest(data))) !=
-          boost::algorithm::to_lower_copy(it->HashString())) {
-        detected_attack = "wrong_hash";
-        return false;
-      }
-    } else if (it->TypeString() == "sha512") {
-      if (boost::algorithm::to_lower_copy(boost::algorithm::hex(Crypto::sha512digest(data))) !=
-          boost::algorithm::to_lower_copy(it->HashString())) {
-        detected_attack = "wrong_hash";
-        return false;
-      }
-    }
-  }
-  detected_attack = "";
-  const bool result = storeFirmware(expected_target_name, data);
-  return result;
-}
```

Comment:
Why get rid of the hash check?

Reply:
I thought we discussed and agreed on it in the chat.

Reply:
Ah yes, true. Yeah, you're right, I think it's pretty redundant. I was trying to think if it added any benefit for testing purposes (which was the original goal of the VirtualSecondary), but now that we have more advanced Secondaries and Secondary test support, it's probably not necessary.